### ISSUE TITLE
Darkmoon Card: Heroism - no chance to criticaly hit

### ISSUE THREAD

**Issue by fidbugs** [elysium-project/server#590]
Darkmoon Card: Heroism(DCH) has no ability to critically hit.

As you can see here, in almost 300 procs not a single one is critical strike.

https://gyazo.com/42a92096e4f6de2b2e9be7ed8b680749

https://gyazo.com/870ac9378742edd2867c68bd2efb6b4d

@GuybrushGit confirmed in comments that currently Herosim has no ability to critical strike.

In this video you can see item being used by rogue in WotLK at time of CATA prepatch came in ZG. On 0:21 it is visible in his combat log that "Heroism" effect critically healed him. It crit heals for 150% of normal value, according to spell critical chance of player. 

https://www.youtube.com/watch?v=pCm-ztUQS_A

Also comment on Allakhazam also states that it can critically heal.

http://web.archive.org/web/20071211130436/http://wow.allakhazam.com:80/db/item.html?witem=19287;source=live

What needs fixing:

a. Adding critical hit(heal) ability at rate of 150% of normal with critical chance equal to spell critical chance of player.
a1. No internal cooldown(if one is present at the moment).

**Comment by killerduki** [comment]
@fidbugs  did you test the proc rate without SoR ?

Probably it is 2% but with SoR it is 4% .

I noticed also weird proc rate with Flurry Axe not being triggered or any other "chance to hit" weapons by SoR .

The only thing i noticed is SoR proc additional Judgements light/wisd procs.

**Comment by fidbugs** [comment]
No i haven't, but it should make no difference. Using SoR would only half the X% proc chance, if SoR does not trigger the proc, not vice versa. 

My calculations take in account both triggering abilities, melee hit and SoR. So if you take out SoR as triggering mechanic, your overall % would increase, not decrease. Using SoR does not increas fixed % of certain chance on hit abilities/items have, it just gives you another ability to trigger it, thus resulting in nominal increase of procs, but the proc chance, X%, remains the same.

I have no info on Flurry Axe, but it should be triggered by SoR, since to my knowledge game takes both SoR and SoC as "melee hits", that's the reason it triggers additional Judgements, so it should trigger everything else that procs on "melee hit".

Also to add up to the bug report, there should be no internal CD on item too.

**Comment by theloras** [comment]
the maelstrom card used to be terrible proc rate too - maybe the Heroism card was as well:

**Darkmoon Card: Maelstrom proc rate is abysmal on Elysium**
https://elysium-project.org/bugtracker/issue/820

**Comment by Mostprovidedlinks** [comment]
There is proof it should be 10%. I will try and find it

**Comment by killerduki** [comment]
@fidbugs  try to test again without SoR for more relevant info , like keep the one you wrote and update it without SoR too.

It will help alot more with 100% accurate info.

**Comment by fidbugs** [comment]
@Mostprovidedlinks Yes that is exactly what i posted from thottbot, it is 10% proc chance.

@theloras Ye i recall that too, tho i still don't know if it is scripted with fixed % chance or with PPM.

@killerduki I tested it today with different weapon, with SoR, proc chance seems to be 9%. Thing is - i am suspecting that it is scripted with PPM mechanic as some enchants are, thus results differ on weapon used, and if SoR is used. Which should not be the case, it should have fix proc chance.

https://gyazo.com/de6973ee57b48995943d0716e5e8e04c

Ill test more tomorrow without SoR, then thing will be more clear - possibility is that SoR is not triggering procs, as i today found out that Judgement is not triggering Maelestorm and Heroism while it triggers Sulfuras and Fiery.

https://gyazo.com/faeea2123190ab4c14d4c7c576a94b75

**Comment by killerduki** [comment]
@fidbugs  now we got to the conclusion , regarding Darkmoon Trinket Maelstrom , there was an Evidence found that it should proc more with slower weapon and proc less with faster weapon. 

This can be case with this Trinket either , since it was proven like that for the other trinket.
I believe the proc rate could be correct and thus this require more investigation.

Remember @theloras  when we reported Maelstrom proc rate , the evidence given was according to speed of your weapon.

**Comment by GuybrushGit** [comment]
FYI Heroism currently has 3 PPM and can crit.

**Comment by fidbugs** [comment]
I did test today just doing regular hits with Sulfuras, 3.6 weapon speed.

Found that proc rate was around (with that setup) 17%, thats exactly what @GuybrushGit says, translates to 3PPM.

https://gyazo.com/870ac9378742edd2867c68bd2efb6b4d

Thing is, not Heroism, not even Maelestorm uses PPM mechanic. In vanilla PPM was reserved for rare things such as enchants(Crusader, Lifesteal, Fiery, etc. and some spells - Seal of Command for instance). PPM on item procs was often confused as mechanic because of before mentioned items and spells that really worked on that mechanic(so called normalization). All regarding PPM and Darkmoon cards was just rumor, due to the lack of ability to track procs, vague explanations, and lack of basic knowledge of game by average player - resulted in a mess that we today had to unfold regarding certain items.

http://www.elitepvpers.com/forum/wow-guides-templates/45008-weapon-enchantment-procs.html - you can see here where person explains PPM mechanic of enchants, but states "no statistical data... rumored PPM 1" - if it was truly PPM item it would ahve been common knowledge same as Crusader, Lifesteal, SoC, etc.

Ill open Malestorm in future on separate topic.

So staying on track here with Heroism, it indeed seems that currently game has it set to 3 PPM, which should be changed fixed 10% chance to proc, add ability to critically hit(150% of normal), remove PPM mechanic, and remove internal CD if it exists at the moment.

**Comment by killerduki** [comment]
@fidbugs 

http://web.archive.org/web/20070706163900/http://www.thottbot.com:80/i19287

Some of comments says that is PPM , some of comments says that as "Duel Wield" it proc 4 times.

You gotta bring pure evidence.

**Comment by fidbugs** [comment]
I have been wrong, kinda fooled by some conflicting evidence(mostly i was missled by SoC, SoR and Judgement for which i was assumed trigger enchants and item procs, which is not fully true, but thats story for another bug report). To cut two day long story short.

You were right, both Malestrom and Heroism use same mechanic as enchants and SoC(tho it is not PPM in sense of TBC PPM, but for the sake of argument and explanation we can call it "soft" PPM or vanilla PPM). Somehow i did not see Malestrom bug report initially and things written there, so thru my investigation regarding Herosim, enchants, etc. i came to same conclusion.

TLDR: 

Current 3PPM is correct, all that need fixing is ability to crit equal to your spell crit chance(and removal of internal CD if one exists).

Explanation( for those who like to read):

This is what i did. I went second by second thru that combat log of rogue in ZG with Heroism, and reverse engineered its mechanic. First assumption was that it works same as enchants, thus game assigns certain chance to proc per hit according to MH weapon speed(which would then result in observed average number of procs in a minute - "soft" PPM). Since it was rogue with almost constant Slice and Dice dual wield, raw number of hits and proc occurrences would not help much. I had to take time spent hitting and number of procs to get "raw" PPM.

There was total of 36 Heroism procs during 6.75 minutes totaling 5.33 PPM. to this number we need to account off hand attacks, special strikes that trigger Heroism - and for that i used combination of my own estimate and same difference tracked on Lifesteal test, which is roughly 0.5 difference in using Slice and Dice and Sinister strike, getting us to number of 2.96 PPM.  

Lifesteal test - http://rogerkarlsson.com/blogs/wow/lifestealing/

There is lot of variables involved here, dual wield, special attacks, Slice and Dice, multi target abilities, not to forget that even that observed number of procs is in rather short period of time on small sample(around 500 hits) so the observed PPM from video could be higher(same as in Lifesteal and Crusader test we get 4.5PPM and 0.8PPM, which are ingame 6PPM and 1PPM - around 20% difference).

For me this is enough of evidence, if you want more feel free to analyze the video more, gl with it :)
